sdk/lib/atl/atlcom.h

```diff
@@ -663,6 +702,24 @@ public:
     class::GetCategoryMap,                                                        \
     class::ObjectMain },
 
+
+
+#define OBJECT_ENTRY_AUTO(clsid, class)                                                                                \
+    ATL::_ATL_OBJMAP_ENTRY __objMap_##class = {                                                                        \
+        &clsid,                                                                                                        \
+        class ::UpdateRegistry,                                                                                        \
+        class ::_ClassFactoryCreatorClass::CreateInstance,                                                             \
+        class ::_CreatorClass::CreateInstance,                                                                         \
+        NULL,                                                                                                          \
+        0,                                                                                                             \
+        class ::GetObjectDescription,                                                                                  \
+        class ::GetCategoryMap,                                                                                        \
+        class ::ObjectMain};                                                                                           \
+    extern "C" _ATLALLOC("ATL$__m") ATL::_ATL_OBJMAP_ENTRY *const __pobjMap_##class = &__objMap_##class;               \
+    OBJECT_ENTRY_PRAGMA(class)
+
+
+
```

Comment:
Three newlines?

Reply:
Yes, between the big ugly macro's to make it a bit clearer what is what.